tuna/miopen/celery_tuning/tuning.py

```diff
@@ -0,0 +1,390 @@
+                                                  library.dbt)
+          serialized_jobs = serialize_chunk(entries)
+
+          #build context for each celery task
+          for job, config in serialized_jobs:
+            context = {
+                'job': job,
+                'config': config,
+                'worker_type': worker_type,
+                'arch': library.dbt.session.arch,
+                'num_cu': library.dbt.session.num_cu,
+                'kwargs': kwargs,
+                'fdb_attr': fdb_attr
+            }
+
+            #calling celery task, enqueuing to celery queue
+            res_set.add(celery_enqueue.apply_async((context,), queue=q_name))
+
+        if not job_list:
+          if not res_set:
+            return False
+          LOGGER.info('All tasks added to queue')
+          break
+      except KeyboardInterrupt:
+        LOGGER.error('Keyboard interrupt caught, draining results queue')
+        session.rollback()
+        stop_active_workers()
+        results_gather(res_set, worker_type)
+        purge_queue([q_name], LOGGER)
+
+  results_gather(res_set, worker_type)
+  stop_active_workers()
+  end = time.time()
+  LOGGER.info("Took {:0>8}".format(str(timedelta(seconds=end - start))))  #pylint: disable=consider-using-f-string
+
+  return True
+
+
+def results_gather(res_set, worker_type):
+  """Start subproc to drain result queue and populate mysql DB"""
+  LOGGER.info('Started drain process')
+  #start draining result_queue in subprocess
+  drain_process = Process(target=drain_queue, args=[worker_type, DBT])
+  drain_process.start()
+
+  LOGGER.info('Gathering async results in callback function, blocking')
+  #waiting on all results to come in
+  _ = res_set.join(callback=result_callback)
+
+  #terminate result_queue drain process with special queue token (NONE,NONE)
+  LOGGER.info('Adding terminatig token to result_queue')
+  result_queue.put([None, None])
+
+  drain_process.join()
+  LOGGER.info('Done writing from result_queue to mySQL')
+
+
+def drain_queue(worker_type, dbt):
+  """Drain results queue"""
+  interrupt = False
+  LOGGER.info('Draining queue')
+  sleep_time = 0
+  wait_limit = 1800  #max wait time
+  seen_res = False
+  with DbSession() as session:
+    while True:
+      try:
+        fin_json, context = result_queue.get(True, 1)
+        #LOGGER.info('Parsing: %s', fin_json)
+        #LOGGER.info('Parsing context: %s', context)
+        if fin_json is None and context is None:
+          LOGGER.info('Reached end of results queue')
+          break
+        if worker_type == 'fin_build_worker':
+          process_fin_builder_results(session, fin_json, context, dbt)
+          seen_res = True
+        else:
+          process_fin_evaluator_results(session, fin_json, context, dbt)
+          seen_res = True
+      except KeyboardInterrupt:
+        if result_queue.empty():
+          return True
+        interrupt = True
+        #continue
```

Comment:
In the case where Tuna is terminated early via interrupt. This would not stop until the result queue is empty and a second keyboard interrupt is issued. Or it could expire from exceeding the wait_limit.
It seems drain_queue relies on wait_limit for it to exit gracefully after Tuna is interrupted.
At the same time wait_limit threatens Tuna's result commits with early termination for long runs.
Would the result queue persist after this process ends? And would it be easy / reliable to restart the enqueueing process to resume committing the results.